Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragmentViewModel.kt

```diff
@@ -281,6 +281,24 @@ class HomeFragmentViewModel @AssistedInject constructor(
             .asLiveData(context = dispatcherProvider.Default)
     }
 
+    data class ShowAdditionalHighRiskLevelDialogEvent(
+        val show: Boolean,
+        val maxEncounterAgeInDays: Int
+    )
+
+    val showAdditionalHighRiskLevelDialog: LiveData<ShowAdditionalHighRiskLevelDialogEvent> by lazy {
```

Comment:
Could this lead to situations where the same dialog is created multiple times and overlays other instances?
E.g. when the livedata reconnects to the flow, like when the user puts the app into background for more than 5 seconds while the dialog is shown.

Reply:
You are right! Thanks for pointing this out. I am using `SingleLiveEvent` now.